### PR TITLE
[WIP] Rustup to 1.9.0-nightly (bf5da36f1 2016-04-06)

### LINKED ISSUE #841: Error: can't find crate for `rustc_front` (latest nightly, 2016-04-06)
```sh
$ rustc --version
rustc 1.9.0-nightly (bf5da36f1 2016-04-06)
```

Compiling `clippy v0.0.61` on the latest nightly fails with:

```sh
   Compiling clippy v0.0.61
... /clippy-0.0.61/src/lib.rs:23:1: 23:26 error: can't find crate for `rustc_front` [E0463]
... /clippy-0.0.61/src/lib.rs:23 extern crate rustc_front;
```

### PR DESCRIPTION
**Pull request by mcarton**
Fix #841. Ref https://github.com/rust-lang/rust/pull/32767.

### CODE REVIEW

File: src/utils/mod.rs [hunk 1]
```diff
@@ -56,7 +56,7 @@ pub const RANGE_TO_PATH: [&'static str; 3] = ["std", "ops", "RangeTo"];
 pub const REGEX_NEW_PATH: [&'static str; 3] = ["regex", "Regex", "new"];
 pub const RESULT_PATH: [&'static str; 3] = ["core", "result", "Result"];
 pub const STRING_PATH: [&'static str; 3] = ["collections", "string", "String"];
-pub const TRANSMUTE_PATH: [&'static str; 3] = ["core", "intrinsics", "transmute"];
+pub const TRANSMUTE_PATH: [&'static str; 4] = ["core", "intrinsics", "", "transmute"];
```
Comment (mcarton) [inline src/utils/mod.rs]: That one is weird.

Comment (Manishearth) [inline src/utils/mod.rs]: I wonder what caused this, and if it's a bug. @eddyb?

Comment (mcarton) [inline src/utils/mod.rs]: I wonder if it could happen otherwise and `match_def_path` needs more work or if it’s just an `intrinsics` weirdness.

Comment (Manishearth) [inline src/utils/mod.rs]: It probably is intrinsics weirdness, but match_def_path is pretty simple so it's likely that the path is just weird. Which might be a "bug" given that paths probably shouldn't work that way.

Comment (eddyb) [inline src/utils/mod.rs]: Turning a `DefPath` into a string still has some weirdnesses, such as not skipping the `extern {...}` item (and some `impl`-related stuff).

**Comment by Manishearth** [comment]
This was merged https://github.com/Manishearth/rust-clippy/commit/76a831c60ff989446016ad0cfc519471661811f1

**Comment by mcarton** [comment]
Oh, a “Verified” tag. Is that the only reason you amended the commit or I am missing something here? :smile:

**Comment by Manishearth** [comment]
Well, I meant to stick it onto the merge commit, but this was an ff merge with no merge commit so it accidentally signed yours instead.

I've started signing my commits and plan to sign clippy releases, perhaps with a short changelog.

**Comment by mcarton** [comment]
> Well, I meant to stick it onto the merge commit, but this was an ff merge with no merge commit so it accidentally signed yours instead.

So soon in the morning, I wasn’t sure I wasn’t missing something :D

> perhaps with a short changelog

I’ve kept a list of rustups and new lints for a while with the intention make a changelog file, but I never did. Let me markdownify that and I’ll make a PR.

**Pull request closed without merging**